src/module/comments/comments.service.ts

```diff
@@ -0,0 +1,114 @@
+import { Injectable, NotFoundException, Post } from '@nestjs/common';
+import { ExceptionCodeName } from 'src/enum/exception-codes.enum';
+import { getRepository } from 'typeorm';
+import { RequestUserPayload } from '../auth/interface/request-user-payload.interface';
+import { UserService } from '../user/user.service';
+import { UnauthorizedException } from '@nestjs/common';
+import { Transactional } from 'typeorm-transactional-cls-hooked';
+import { Comment } from './comments.entity';
+import { UpdateCommentDto } from './dto/update-comment.dto';
+import { CreateCommentDto } from './dto/create-comment.dto';
+import { Post as PostModel} from '../posts/posts.entity';
+import { PostsService } from '../posts/posts.service';
+
+@Injectable()
+export class CommentsService {
+  constructor(
+    private postsService: PostsService,
+    private userService: UserService,
+  ){}
+
+  @Transactional()
+  async create(
+    id: number,
+    createCommentDto: CreateCommentDto,
+    requestUserPayload: RequestUserPayload,
+  ): Promise<Comment>{
+    const user = await this.userService.getOne({
+      where: {
+        id: requestUserPayload.id,
+      },
+    });
+    if (!user) {
+      throw new UnauthorizedException(ExceptionCodeName.INVALID_CREDENTIALS);
+    }
+    
+    const { content, date } = createCommentDto;
+
+    const comment = new Comment();
+    comment.content = content;
+    comment.user = user;
+    date && (comment.created_at = date);
+    const res = await getRepository(Comment).save(comment);
+    const post = await this.postsService.getById(id);
+    post.comments.push(comment);
+    await getRepository(PostModel).save(post);
```

Comment:
No need for pushing to post.comments. If entities are correctly linked, typeorm should automatically join them.